File: Assets/Code/Outfit7/Raven/TrackComponents/AnimationData/CurveSpecialization/RavenAnimationDataCurveBool.cc
#ifdef STARLITE
#include "RavenAnimationDataCurveBool.h"
#include "RavenAnimationDataCurveBool.cs"

namespace Starlite {
    namespace Raven {
        RavenAnimationDataCurveBool::RavenAnimationDataCurveBool() {
        }

        bool RavenAnimationDataCurveBool::EvaluateAtTime(double time, double duration) {
            return m_Curves[0]->Evaluate((float)GetNormalizedTime(GetCurrentTime(time, duration), duration, m_EaseType)) >= 1.f ? true : false;
        }

        void RavenAnimationDataCurveBool::SetStartingValues(const bool& values) {
            m_Curves = Array<Ref<AnimationCurve>>(1);
            m_Curves[0] = Object::New<AnimationCurve>();
            m_Curves[0]->AddKeyframe(AnimationCurve::Keyframe(0, values ? 1.f : 0.f));
            m_Curves[0]->AddKeyframe(AnimationCurve::Keyframe(1, values ? 1.f : 0.f));
        }

        void RavenAnimationDataCurveBool::SyncStartingValues(const bool& values) {
            auto& curve = m_Curves[0];
            auto& key = curve->GetKeyframes()[0];
            key.value = values ? 1.f : 0.f;
        }
    } // namespace Raven
} // namespace Starlite
#endif

File: Assets/Code/Outfit7/Raven/TrackComponents/AnimationData/CurveSpecialization/RavenAnimationDataCurveBool.cs
//{ "engine_register_objects":["Starlite__Raven__RavenAnimationDataCurveBool"] }
#if !__cplusplus
#pragma warning disable
namespace Starlite {

    namespace Raven {
    
        // RavenAnimationDataCurveBool
        [Starlite.ObjectClassAttribute("Starlite::Raven::RavenAnimationDataCurveBool")]
        public sealed partial class RavenAnimationDataCurveBool : global::Starlite.Raven.RavenAnimationDataCurve<bool> {
        }
        
    }
    
}
#pragma warning restore
#else
// RavenAnimationDataCurveBool
namespace Starlite { namespace Raven { 
SIMPLEMENT_CLASS_WITH_PROPERTIES_IN_CLASS_SCOPED(Starlite::Raven::RavenAnimationDataCurveBool, RavenAnimationDataCurveBool, RavenAnimationDataCurveBool, Starlite::Raven::RavenAnimationDataComponentBase, Starlite::SceneObjectComponent, "")
SIMPLEMENT_PROPERTY_UNIVERSAL(RavenAnimationDataCurveBool, m_Curves, m_Curves, kArray, kObject, Starlite::AnimationCurve::TypeId, "")
SIMPLEMENT_PROPERTY_UNIVERSAL(RavenAnimationDataCurveBool, m_EaseType, m_EaseType, kInt, kUnknown, -1, "")
SIMPLEMENT_PROPERTY_UNIVERSAL(RavenAnimationDataCurveBool, m_ValueType, m_ValueType, kInt, kUnknown, -1, "")
SIMPLEMENT_PROPERTY_UNIVERSAL(RavenAnimationDataCurveBool, m_RepeatCount, m_RepeatCount, kInt, kUnknown, -1, "")
SIMPLEMENT_PROPERTY_UNIVERSAL(RavenAnimationDataCurveBool, m_Mirror, m_Mirror, kBool, kUnknown, -1, "")
SIMPLEMENT_CLASS_END(RavenAnimationDataCurveBool)
} } 
SIMPLEMENT_CLASS_SCOPE(Starlite__Raven__RavenAnimationDataCurveBool, Starlite::Raven, RavenAnimationDataCurveBool)
#endif
